pyanaconda/modules/common/structures/subscription.py

```diff
@@ -78,3 +81,281 @@ def addons(self) -> List[Str]:
+        :rtype: str
+        """
+        return self._server_proxy_user
+
+    @server_proxy_user.setter
+    def server_proxy_user(self, username: Str):
+        self._server_proxy_user = username
+
+    # private data
+    # - generally sensitive data such as passwords
+    #   or activation keys
+    # - these values should be "write only",
+    #   meaning data goes to the Subscription
+    #   module but can't be read out later one
+    #   via the public API
+    # - only the Subscription module should have
+    #   have access to these data internally &
+    #   use them appropriately (eq. register
+    #   the system, authenticate to HTTP proxy, etc.)
+    # - it should be also possible to explicitly
+    #   clear a previously set secret
+    # - to protect these values we are using
+    #   SecretData & SecretDataList, see their
+    #   implementation for more information
+
+    @property
+    def account_password(self) -> SecretData:
+        """Red Hat account password.
+
+        NOTE: This property is stored in SecretData
+              nested DBus structure to protect its contents.
+
+        :return: Red hat account password stored in a SecretData instance
+        :rtype: SecretData instance
+        """
+        return self._redhat_account_password
+
+    @account_password.setter
+    def account_password(self, password: SecretData):
+        if password:
+            self.account_password_set = True
+        self._redhat_account_password = password
+
+    @property
+    def activation_keys(self) -> SecretDataList:
+        """List of activation keys.
+
+        For a successful activation key based registration
+        at least one activation key needs to be set.
+
+        NOTE: This property is stored in SecretDataList
+              nested DBus structure to protect its contents.
+
+        :return: list of activation keys stored in SecretDataList instance
+        :rtype: SecretDataList instance
+        """
+        return self._activation_keys
+
+    @activation_keys.setter
+    def activation_keys(self, activation_keys: SecretDataList):
+        self._activation_keys = activation_keys
+
+    @property
+    def server_proxy_password(self) -> SecretData:
+        """RHSM HTTP proxy - access password.
+
+        NOTE: This property is stored in SecretData
+              nested DBus structure to protect its contents.
+
+        :return: RHSM HTTP proxy password stored in SecretData instance
+        :rtype: SecretData instance
+        """
+        return self._server_proxy_password
+
+    @server_proxy_password.setter
+    def server_proxy_password(self, password: SecretData):
+        self._server_proxy_password = password
+
+    def __repr__(self):
+        """Provide a string representation of SubscriptionRequest."""
+        # we don't need to skip the sensitive data as the
+        # SecretData & SecretDataList classes handle that
+        # automatically
+        return generate_string_from_data(self)
```

Comment:
This is the default implementation. You can remove it.

Reply:
Oh, cool, that's handy. :)